File: 0566/main.py
from typing import List


class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        if len(mat) == 0 or len(mat[0]) == 0 or r * c != len(mat) * len(mat[0]):
            return mat
        
        res: List[List[int]] = []
        newRowArray: List[int] = []
        column = 0

        for rowArray in mat:
            for value in rowArray:
                if column < c:
                    newRowArray.append(value)
                    column += 1
                else:
                    res.append(newRowArray)
                    newRowArray = [value]
                    column = 1
        res.append(newRowArray)
        return res

File: 0566/main.go
package main

func matrixReshape(mat [][]int, r int, c int) [][]int {
	if len(mat) == 0 || len(mat[0]) == 0 || r*c != len(mat)*len(mat[0]) {
		return mat
	}

	res := [][]int{}
	newRowArray := []int{}
	column := 0

	for _, rowArray := range mat {
		for _, value := range rowArray {
			if column < c {
				newRowArray = append(newRowArray, value)
				column += 1
			} else {
				res = append(res, newRowArray)
				newRowArray = []int{value}
				column = 1
			}
		}
	}
	return append(res, newRowArray)
}

func main() {
	matrixReshape([][]int{
		{1, 2}, {3, 4},
	}, 1, 4)
}
